webapp/Sigbuilder-graph.js

```diff
@@ -0,0 +1,1872 @@
+            var reader = new FileReader();
+            reader.onload = function(progressEvent){
+            // By lines
+                var lines = this.result.split('\n');
+                for(var i = 0; i < lines.length; i++){
+                    var line = lines[i];
+                    if(line.length != 0){
+                            points_ary[i] = scan(line.trim(),format);
+                        if((points_ary[i][0]!=""||points_ary[i][0]!="undefined")&&(points_ary[i][1]!=""||points_ary[i][1]!="undefined")){
+                            sigbuilder_Graph.series[0].addPoint([points_ary[i][0],points_ary[i][1]]);
+                        }
+                    }
+                }
+                graphParameters.mtd = 1;
+                graphParameters.graphPoints = points_ary.slice();
+                pointsHistory.push(graphParameters.graphPoints.slice());
+                var pointscount = sigbuilder_Graph.series[0].data.length;
+                var xmaxtitle = (sigbuilder_Graph.xAxis[0].getExtremes().dataMax).toFixed(6);
+                sigbuilder_Graph.setTitle(null, { text: updateSubtitleForSigbuilderGraph(pointscount, graphParameters.mtd, xmaxtitle, graphParameters.PeriodicOption)});
+                sigbuilder_Graph.series[0].update({
+                type: "line",
+                step: '',
+                name: ''
+                });
+                autoscaleFunctionalityForGraph(sigbuilder_Graph,graphParameters);
+            };
+            reader.readAsText(file);
+        }else{
+            document.getElementById("messageLabel").innerHTML ="Bad Format";
+        }
+        wind.destroy();
+    };
+
+}
+
+function saveToTextFile(graph,sigbuilder_Graph){
+
+    // Create basic structure for the form
+    var content = document.createElement('div');
+    content.setAttribute("id", "saveToDatFile");
+
+    // Add Form
+    var myform = document.createElement("form");
+    myform.method = "post";
+    myform.id = "formSaveToDatFile";
+    myform.style.padding = "10px";
+
+    var titlelabel = document.createElement('span');
+    titlelabel.innerHTML = "Text data file";
+    myform.appendChild(titlelabel);
+    // Line break
+    var linebreak = document.createElement('br');
+    myform.appendChild(linebreak);
+    // Line break
+    var linebreak = document.createElement('br');
+    myform.appendChild(linebreak);
+
+    // Line break
+    var linebreak = document.createElement('br');
+    myform.appendChild(linebreak);
+
+    var labelArray = ['Filename','Writing [C] format'];
+    var textValueArray = ["mydatafile.dat","%g %g"];
+    for(var i = 0; i < labelArray.length; i++){
+        // Input Title
+        var namelabel = document.createElement('label');
+        namelabel.innerHTML = labelArray[i];
+        namelabel.style.marginLeft = "30px";
+        myform.appendChild(namelabel);
+
+        // Input
+        var input = document.createElement("input");
+        input.name = "edit_"+labelArray[i].toString();
```

Comment:
```suggestion
        input.name = "edit_"+labelArray[i];
```
labelArray is a string array. don't use toString here.